src/init.cc

```diff
@@ -1735,6 +1735,7 @@ ncclResult_t ncclCommInitAll(ncclComm_t* comms, int ndev, const int* devlist) {
     for (int i = 0; i < ndev; ++i) {
       /* invalid device check. */
       if (devlist[i] < 0 || devlist[i] >= totalnDev) {
+        WARN("Invalid device %d (totalnDev=%d)", devlist[i], totalnDev);
```

Comment:
Not sure how this is related to the commit. 

It's indeed a valid issue (and we should return ncclInvalidArgument, not ncclUnhandledCudaError btw). I guess nobody uses ncclCommInitAll any more which is why nobody noticed so far.